### ISSUE TITLE
Undocumented = operator syntax in Generic Constraining

### ISSUE THREAD

**Issue by Ethan-Arrowood** [Microsoft/TypeScript#30337]
In the declaration file for Fastify (https://github.com/fastify/fastify/blob/master/fastify.d.ts) there is some TypeScript code I do not recognize and cannot find another example of in the TypeScript documentation:
```typescript
declare function fastify<
  HttpServer extends (http.Server | http2.Http2Server) = http.Server,
  HttpRequest extends (http.IncomingMessage | http2.Http2ServerRequest) = http.IncomingMessage,
  HttpResponse extends (http.ServerResponse | http2.Http2ServerResponse) = http.ServerResponse
>(opts?: fastify.ServerOptions): fastify.FastifyInstance<HttpServer, HttpRequest, HttpResponse>;
```
I recognize the Generic constraint syntax with the `extends` keyword, but what about the `= http.Server` part? Is this like a default parameter or something?

Whatever this syntax means it should be better documented.

I tried googling for this syntax by searching 
- `typescript generic constraint`
- `typescript generic constraint assignment`
- `typescript generic defaults`

**Comment by RyanCavanaugh** [comment]
I agree this should be covered by the handbook, though it is discoverable through the search you tried

![image](https://user-images.githubusercontent.com/6685088/54217428-d62d5380-44a8-11e9-998d-655259aa5501.png)

![image](https://user-images.githubusercontent.com/6685088/54217455-e1807f00-44a8-11e9-922e-2a802324343a.png)

**Comment by Ethan-Arrowood** [comment]
Ahh i see. I was looking at the wrong documentation. I think what deterred me from this page originally was that it was v2.3 not v3. I don't like to read different major version docs because it can have unsupported information

**Comment by rossng** [comment]
It's near impossible to find out what this syntax does if you haven't already guessed that it's for generic parameter defaults. The syntax is un-Googleable. I had to use an AST browser to figure out what was going on!

A vote from me for this to be documented on the [Generics](https://www.typescriptlang.org/docs/handbook/generics.html) page.

**Comment by HartS** [comment]
4 years after this syntax was introduced, I had to ask in the Typescript discord to get an answer to this. As mentioned by @rossng, completely impossible to google, or find from the TS docs.
